### ISSUE TITLE
TEP-0700 New PHY vs Old PHY

### ISSUE THREAD

**Issue by JeanPhilippeLebel** [TechNexion/meta-tn-imx-bsp#28]
Hello,

We have a wide range of TEP-0700 systems with some of them using the old PHY and, of course, a bunch with the new PHY.

During testing, we noticed that systems with the new PHY do not work properly (no network). The same image works correctly on older systems.

Any idea why that would append?

**Comment by ray-chang** [comment]
@JeanPhilippeLebel 

I don't know the details of the product you got and the version of the software you used. This is the software migration instructions page, where you can check if the kernel you are using has the relevant patches.

https://developer.technexion.com/docs/software-migration-for-products-with-realtek-phy

By the way, if you have any questions about the product, please contact TechNexion FAE or Sales.

BR,
Ray

**Comment by JeanPhilippeLebel** [comment]
Ok, I'm back with this. 

The document suggested is for the older kernel versions. 

I've tested with imx-6.1.55-2.2.0 (https://github.com/TechNexion/tn-imx-yocto-manifest/blob/mickledore_6.1.y-stable/imx-6.1.55-2.2.0.xml), without any modifications. 

Using a TEP-0700 with the older PHY, I can get a network connection. With the newer PHY (realtek), I can't get an IP using DHCP. The PHY does detect if the cable is connected, or not, but nothing more. 

I've compared the code of drivers/net/phy/realtek.c (between t n-imx_5.15.71_2.2.0 and tn-imx_5.10.72_2.2.0-stable) it seems some code was left out. The missing code was introduced by this commit https://github.com/TechNexion/linux-tn-imx/commit/131ea068cefa25dbebf4e3fea0fd546867d4e66a

Thanks
